File: how-to-use-kotlin-in-java-project/src/main/java/dev/trile/customexception/VietcomBank.java
package dev.trile.customexception;

import static dev.trile.customexception.ErrorType.TRANSFER_MONEY_NOT_ENOUGH_MONEY;

public class VietcomBank implements Bank {

  @Override
  public void transferMoney(String accountNo, int amount) throws SalaryException {
    checkTransferParams(accountNo, amount);

    if (amount > 10_000_000)
      throw new SalaryException(TRANSFER_MONEY_NOT_ENOUGH_MONEY, "VCB only support transfer amount less than 10M");

    System.out.println("VCB transfer success");
  }
}

File: how-to-use-kotlin-in-java-project/src/main/kotlin/dev/trile/java2kotlin/VietcomBank.kt
package dev.trile.java2kotlin

import dev.trile.customexception.Bank
import dev.trile.customexception.ErrorType
import dev.trile.customexception.SalaryException

class VietcomBank : Bank {

  @Throws(SalaryException::class)
  override fun transferMoney(accountNo: String, amount: Int) {
    checkTransferParams(accountNo, amount)

    if (amount > 10000000)
      throw SalaryException(ErrorType.TRANSFER_MONEY_NOT_ENOUGH_MONEY, "VCB only support transfer amount less than 10M")

    println("VCB Kotlin transfer success")
  }
}
